ocaml/idl/datamodel_tracing.ml

```diff
@@ -0,0 +1,184 @@
+(*
+ * Copyright (C) 2023 Cloud Software Group
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation; version 2.1 only. with the special
+ * exception on linking described in file LICENSE.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *)
+
+open Datamodel_types
+open Datamodel_common
+open Datamodel_roles
+
+let create =
+  call ~name:"create" ~in_oss_since:None ~lifecycle:[]
+    ~doc:"Set the fields of a TracerProvider in the Tracing Library"
+    ~params:
+      [
+        (String, "name_label", "The TracerProvider")
+      ; ( Set (Ref _host)
+        , "hosts"
+        , "Hosts the TracerProvider is to be created on"
+        )
+      ; (Bool, "enabled", "The enabled of the TracerProvider")
+      ; (Map (String, String), "tags", "The tags of the TracerProvider")
+      ; (Set String, "endpoints", "The endpoints of the TracerProvider")
+      ; (Set String, "components", "The componenets of the TracerProvider")
+      ; (Set String, "filters", "The filters of the TracerProvider")
+      ; (Set String, "processors", "The processors of the TracerProvider")
+      ]
+    ~allowed_roles:_R_POOL_ADMIN ()
+
+let destroy =
+  call ~name:"destroy" ~in_oss_since:None ~lifecycle:[]
+    ~doc:"Set the fields of a TracerProvider in the Tracing Library"
+    ~params:[(Ref _tracing, "self", "The TracerProvider to be destoryed")]
+    ~allowed_roles:_R_POOL_ADMIN ()
+
+let set_hosts =
+  call ~name:"set_hosts" ~in_oss_since:None ~lifecycle:[]
+    ~doc:"Set the host field of a TracerProvider in the Tracing Library"
+    ~params:
+      [
+        (Ref _tracing, "self", "The TracerProvider")
+      ; ( Set (Ref _host)
+        , "hosts"
+        , "Hosts the TracerProvider is to be created on"
+        )
+      ]
+    ~allowed_roles:_R_POOL_ADMIN ()
+
+let set_enabled =
+  call ~name:"set_enabled" ~in_oss_since:None ~lifecycle:[]
+    ~doc:"Set the enabled field of a TracerProvider in the Tracing Library"
+    ~params:
+      [
+        (Ref _tracing, "self", "The TracerProvider")
+      ; ( Bool
+        , "enabled"
+        , "The enabled the TracerProvider is to be set to. Enabled (true) or \
+           disabled (false) this TraceProvider.t"
+        )
+      ]
+    ~allowed_roles:_R_POOL_ADMIN ()
+
+let set_tags =
+  call ~name:"set_tags" ~in_oss_since:None ~lifecycle:[]
+    ~doc:"Set the tags field of a TracerProvider in the Tracing Library"
+    ~params:
+      [
+        (Ref _tracing, "self", "The TracerProvider")
+      ; ( Map (String, String)
+        , "tags"
+        , "The tags that are emitted to spans derived from this provider"
```

Comment:
I like this description much better, this explains to the users how are the tags going to be used for